aspnet/fundamentals/localization.rst

```diff
@@ -1,8 +1,285 @@
+- Resources/ViewModels/Account/RegisterViewModel.fr.resx
+  
+.. literalinclude:: localization/sample/ViewModels/Account/RegisterViewModel.cs
+  :language: c#
+  :lines: 9-26
+  :dedent: 3
+  
+Currently, the run time doesn't look up localized strings for non-validation attributes. In the code above, the "Email" (from ``[Display(Name = "Email")]``) will not be localized.
+  
+Localization middleware
+-----------------------------------
+
+The current culture on a request is set in the localization :doc:`/fundamentals/middleware`. The localization middleware is enabled in the ``Configure`` method of *Startup.cs* file.
+
+.. literalinclude:: localization/sample/StartupConfig.cs
+  :language: c#
+  :linenos:
+  :lines: 106-107, 136-159
+  :dedent: 6
+
+``UseRequestLocalization`` initializes a ``RequestLocalizationMiddleware`` object. On every request the list of ``RequestCultureProviders`` in the ``RequestLocalizationOptions`` is enumerated and the first non-null provider is used. The default providers come from the ``RequestLocalizationOptions`` class:
+
+#. ``QueryStringRequestCultureProvider``
+#. ``CookieRequestCultureProvider``
+#. ``AcceptLanguageHeaderRequestCultureProvider``
+
+The default list goes from most specific to least specific. Later in the article I'll show how you can change the order and even add a custom localization provider. If there are no non-null providers, the ``DefaultRequestCulture`` is used.
+    
+QueryStringRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In addition the providing the primary mechanism some app will use to set the culture and UI culture, adding a query string to the URL is useful for debugging and testing code. Unless you change the ``RequestCultureProviders`` list, a query string to the will always win as the localization provider. You pass the query string parameters ``culture`` and ``ui-culture``. The following example sets the specific culture (language and region) to Spanish/Mexico:
+
+``http://localhost:5000/?culture=es-MX&ui-culture=es-MX``
+
+If you only pass in one of the two (``culture`` or ``ui-culture``), the query string provider will set both values using the one you passed in. For example, setting just the culture will set both the ``Culture`` and the ``UICulture``:
+
+``http://localhost:5000/?culture=es-MX``
+
+CookieRequestCultureProvider
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Production apps will often provide a mechanism to set the culture with the ``ASPNET_CULTURE`` cookie. The sample app provides code to set the cookie for French and English cultures.
+
+The Accept-Language HTTP header
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Accept-Language HTTP header from a browser request is not an infallible way to detect the user's preferred language. This setting indicates what the browser has been set to send or has inherited from the underlying operating system. A production app should include a way for a user to customize their choice of culture.
```

Comment:
Before jumping into what the Accept-Language header is not add an initial description of what it is and what it specifies. I believe you have some text later in the doc that describes the format of this header that should probably go here, or just link to the header definition in the HTTP spec.